src/providers/cambridge/test_utils.py

```diff
@@ -0,0 +1,32 @@
+import pytest
+from utils import get_page_content, find_links_and_tags, download_file
+from bs4 import BeautifulSoup
+import os 
+
+@pytest.mark.vcr()
+def test_page_content_valid():
+    response = get_page_content('https://www.cambridge.org/core/services/librarians/kbart')
+    assert response.status_code == 200
+
+@pytest.mark.vcr()
+def test_page_content_invalid():
+    response = get_page_content('https://www.cambridge.org/core/services/librarians/kbartWontWork')
+    assert response.status_code == 404 
+
+@pytest.mark.vcr
+def test_found_links():
+    url = 'https://www.cambridge.org/core/services/librarians/kbart'
+    response = get_page_content(url)
+    soup = BeautifulSoup(response.text, 'html.parser')
+    found_links = find_links_and_tags(soup, ['computer science'], 'cambridge ebooks and partner presses: 2023 ')
+    assert not len(found_links) == 0
```

Comment:
Please, put how many links are actually found, instead of: `not len`, makes it more precise test

Also, can you add a test, where is 0 links found? You can simulate that response, just read it from a file. Let's say copy the xml to the file and remove the tags you are looking for. And the you can use this file for a test input.